elmslie-plugin/src/main/kotlin/vivid/money/elmslie/plugin/core/ext/PsiDirectoryExt.kt

```diff
@@ -0,0 +1,19 @@
+package vivid.money.elmslie.plugin.core.ext
+
+import com.intellij.psi.PsiDirectory
+
+private const val DOT = "."
+
+fun PsiDirectory.formPackageName(): String {
+    val result = StringBuilder(this.name)
+    var parent: PsiDirectory = this.parent ?: return result.toString()
+
+    while (parent.name != "java") {
```

Comment:
there could also be a `kotlin` source set

Reply:
added